rust/automerge/src/automerge.rs

```diff
@@ -1309,6 +1345,64 @@ impl ReadDoc for Automerge {
         Ok(buffer)
     }
 
+    fn get_marks<O: AsRef<ExId>>(&self, obj: O) -> Result<Vec<Mark<'_>>, AutomergeError> {
+        let (obj, obj_type) = self.exid_to_obj(obj.as_ref())?;
+        let encoding = ListEncoding::new(obj_type, self.text_encoding);
+        let ops_by_key = self.ops().iter_ops(&obj).group_by(|o| o.elemid_or_key());
+        let mut pos = 0;
+        let mut marks = MarkStateMachine::default();
+
+        Ok(ops_by_key
```

Comment:
I think this could be done without the side-effecting `pos` stuff by using a `fold::<(usize, Vec<Mark>), ..>((0, Vec::new(), ..)` rather than than a `filter_map` followed by `collect`